joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/astcreation/AstForDeclarationsCreator.scala

```diff
@@ -352,7 +326,35 @@ trait AstForDeclarationsCreator { this: AstCreator =>
     }
     diffGraph.addNode(createDependencyNode(name, referenceName, IMPORT_KEYWORD))
     createImportNodeAndAttachToAst(impDecl, referenceName, name)
-    Ast()
+    astForRequireCallFromImport(name, referenceName, impDecl)
+  }
+
+  private def astForRequireCallFromImport(name: String, from: String, nodeInfo: BabelNodeInfo): Ast = {
```

Comment:
this doesn't seem to handle [named imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#description
) yet.

Also, do we have to do hoist this to the start of `:program`? I remember @ml86 saying that imports work at any level, but I'm not sure to what extent our dynamic type engine cares about statement order.

Reply:
Named imports are all handled, see the test cases. 
The resulting AST is added to the exact same position where the import decl itself would appear.

Reply:
I don't see any field access calls here, so the handling can't be correct.

Reply:
https://github.com/joernio/joern/blob/4f15589e72e982642aef6fc59412131a37178ada/joern-cli/frontends/jssrc2cpg/src/test/scala/io/joern/jssrc2cpg/passes/ast/DependencyAstCreationPassTest.scala#L159